framework/config/config.py

```diff
@@ -72,6 +72,13 @@ def initialize_attributes(self):
         for type in CONFIG_TYPES:
             self.Config[type] = {}
 
+    def select_config_path(self, file_path):
```

Comment:
The name should be more explicit, such as `user_path_or_default`. Also there should be a docstring to explain what is does, what it returns, etc.